dev-util/visual-studio-code/visual-studio-code-1.22.2.ebuild

```diff
@@ -0,0 +1,98 @@
+# Copyright 1999-2018 Gentoo Foundation
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=6
+
+inherit eutils xdg-utils
+
+DESCRIPTION="Microsoft's free and open-source code editor."
+HOMEPAGE="https://code.visualstudio.com/"
+SRC_URI="https://github.com/Microsoft/vscode/archive/${PV}.tar.gz -> ${P}.tar.gz
+	https://www.hlmjr.com/vscode-assets.tar.gz"
+
+LICENSE="MIT"
+SLOT="0"
+KEYWORDS="~amd64 ~x86"
+IUSE=""
+
+DEPEND="net-libs/nodejs
+	sys-apps/yarn"
+RDEPEND="${DEPEND}
+	dev-libs/nss
+	dev-vcs/git
+	gnome-base/gconf
+	media-libs/alsa-lib
+	x11-libs/gtk+:2
+	x11-libs/libnotify
+	x11-libs/libxkbfile
+	x11-libs/libXScrnSaver
+	x11-libs/libXtst"
+
+S="${WORKDIR}"/vscode-"${PV}"
+
+pre_build_checks() {
+	CHECKREQS_MEMORY="2G" # due to max_old_space_size node setting below
```

Comment:
where is `inherit check-reqs` ?
Check out eclass docs how to use it.
https://devmanual.gentoo.org/eclass-reference/check-reqs.eclass/index.html
You never call the check itself, just define a custom function.

Also the check probably should not be carried if merge type is binary.
check `dev-python/pypy3/pypy3-6.0.0.ebuild` how to do that.